### ISSUE TITLE
TagMacro: invalid conversion from `TagK3[F]` to `TagKK[F[Any, ?, ?]]`

### ISSUE THREAD

**Issue by neko-kai** [zio/izumi-reflect#83]
Given:

```scala
trait BIO3[F[-_, +_, +_]]
type BIO[F[+_, +_]] = BIO3[Lambda[(`-R`, `+E`, `+A`) => F[E, A]]]

def direct[F[+_, +_]: TagKK] =
  Tag[BIO[F]].tag

def indirectFrom3[F[-_, +_, +_]: TagK3] = 
  direct[F[Any, +?, +?]]

println(indirectFrom3[zio.ZIO])
println(direct[zio.ZIO[Any, +?, +?]])
```

Results in `indirectFrom3` returning an invalid, mangled tag:

```
BIO3[=λ %1,%2,%3 → ZIO[-Any,+3,+3]]
```

while `direct` is correct:

```
BIO3[=λ %1,%2,%3 → ZIO[-Any,+2,+3]]
```

The difference here is between the `TagKK` generated from `TagK3` in `indirectFrom3` and in direct `TagKK` in `direct`. Their printouts also differ:

```scala
def tag1[F[-_, +_, +_]: TagK3] = TagKK[F[Any, +?, +?]].tag
def tag2[F[-_, +_, +_]](implicit t: TagKK[F[Any, +?, +?]]) = TagKK[F[Any, +?, +?]].tag

println(s"tag1: ${tag1[zio.ZIO]}")
println(s"tag2: ${tag2[zio.ZIO]}")
```

```
tag1: λ %1,%2 → ZIO[-Any,+1,+2]
tag2: λ %0,%1 → ZIO[-Any,+0,+1]
```

/cc @pshirshov

**Comment by pshirshov** [comment]
https://github.com/zio/izumi-reflect/blob/develop/izumi-reflect/izumi-reflect/src/test/scala-2/izumi/reflect/test/LightTypeTagProgressionTest.scala#L43

**Comment by pshirshov** [comment]
Fixed in a2225377a45a3ff298dcddbfa3f7bd070dde9342
